backend/server/src/controllers/tournamentController.ts

```diff
@@ -24,27 +36,35 @@ export class TournamentController extends Controller {
     return await this.service.createTournament(tournamentData);
   }
 
-  // // @Security("jwt")
-  // @Get("all")
-  // @Tags("Tournament")
-  // public async getAllTournaments(): Promise<Tournament[]> {
-  //     console.log("getAllTournaments TOURNAMENT");
-  //     return await this.service.getAllTournaments();
-  // }
-
-  // // @Security("jwt")
-  // @Get("running")
-  // @Tags("Tournament")
-  // public async getRunningTournaments(): Promise<Tournament[]> {
-  //     return await this.service.getRunningTournaments();
-  // }
-
-  // // @Security("jwt")
-  // @Get("upcoming")
-  // @Tags("Tournament")
-  // public async getUpcomingTournaments(): Promise<Tournament[]> {
-  //     return await this.service.getUpcomingTournaments();
-  // }
+  @Security("jwt")
+  @Put("{tournamentId}/addPlayer")
+  @Tags("Tournament")
+  @Response(404, "Not Found")
+  @Response(400, "Bad Request")
+  @Response(500, "Internal Server Error")
+  public async addPlayerToTournament(
+    @Path() tournamentId: ObjectIdString,
+    @Body() requestBody: AddPlayerRequest
+  ): Promise<Tournament | { message: string }> {
+    try {
+      const result = await this.service.addPlayerToTournament(
+        tournamentId,
+        requestBody.playerId
+      );
+      this.setStatus(200); // OK status
+      return result;
+    } catch (error) {
+      if (error instanceof NotFoundError) {
+        this.setStatus(404); // Not Found
+      } else if (error instanceof Error) {
+        this.setStatus(400); // Bad Request or other appropriate status
+      } else {
+        this.setStatus(500);
+      }
+      const errObj = error as Error;
+      return { message: errObj.message };
+    }
```

Comment:
This is handled in the error handler already. No need to add any logic to the controller layer.